File: HexChat Downloader/program/objects/Bot.py
"""
Bot
Class that models a bot

Created on May 7, 2015
Modified on May 7, 2015

@author Hermann Krumrey
@version 0.1
"""

#imports

"""
Bot
The class that models a bot
"""
class Bot(object):
    
    """
    Constructor
    constructs a new Bot object
    @param botName - the bot's name
    @param serverName - the server's name on which the bot is reacheable
    @param channelName - the channel's name on which the bot is reacheable
    """
    def __init__(self,botName,serverName,channelName):
        self.botName = botName
        self.serverName = serverName
        self.channelName = channelName

File: Objects/Bot.cpp
/*
 * @author Hermann Krumrey <hermann@krumreyh.com>
 */
#include "ServerList.h"


//public
//Constructor

/**
 * Creates a new Bot object with a given name
 * @param name - the name of the bot
 */
Bot::Bot(string name) {

    this->name = name;

}

/**
 * Adds this bot to a given channel
 * @param channel - the channel to which this bot is to be added
 */
void Bot::addToChannel(Channel channel) {

    this->channel.push_back(channel);

}

/**
 * Adds a pack to this bot
 * @param pack - the pack to be added
 */
void Bot::addPack(Pack pack) {

    pack.addToBot(*this);
    this->packs.push_back(pack);

}


//Getter/Setter

string Bot::getName() {
    return this->name;
}

vector<Pack> Bot::getPacks() {
    return this->packs;
}

Channel Bot::getChannel() {
    return this->channel[0];
}